src/Appointments/Domain/Actions/ValidateClinicDoctorRelation.php

```diff
@@ -12,7 +13,7 @@ class ValidateClinicDoctorRelation
     public function execute(AppointmentDto $appointmentDto): bool
     {
         return Doctor::whereId($appointmentDto->doctor_id)
-            ->whereHas('clinics', fn ($q) => $q->whereClinicId($appointmentDto->clinic_id))
+            ->whereHas('clinics', fn (Builder $q) => $q->whereClinicId($appointmentDto->clinic_id))
```

Comment:
I think you're trying to do something like this

```suggestion
            ->whereRelation('clinics', 'id', $appointmentDto->clinic_id)
```